src/auto_ir_metadata/__init__.py

```diff
@@ -0,0 +1,81 @@
+import json
+import zipfile
+from pathlib import Path
+from typing import Optional
+
+from py_measure import Environment as PyMeasureEnvironment
+
+import auto_ir_metadata.utils as utils
+
+
+class Environment(PyMeasureEnvironment):
+    pass
+
+
+def persist_ir_metadata(
+    output_directory: Path,
+    system_name: Optional[str] = None,
+    system_description: Optional[str] = None,
+    environment: Optional[PyMeasureEnvironment] = None,
+):
+    if output_directory and isinstance(output_directory, str):
+        output_directory = Path(output_directory)
+
+    utils._fail_if_parameters_have_wrong_types(system_name, system_description, environment)
+    utils.__ensure_output_directory_is_valid(output_directory)
+    output_file = output_directory / utils.FILE_NAME
+    collected_meta_data = utils.get_python_info()
+
+    if not environment:
+        environment = Environment(["git"])
+        environment.start_measuring()
+        environment.stop_measuring()
+
+    if len(environment.measurements) > 0:
+        for k, v in environment.measurements[-1].items():
+            collected_meta_data[k] = v
```

Comment:
This should work right?
```suggestion
        collected_meta_data.update(environment.measurements[-1])
```

Reply:
Agreed, I resolved it.